jans-cedarling/cedarling/src/common/policy_store/entity_parser.rs

```diff
@@ -0,0 +1,686 @@
+            .parents
+            .iter()
+            .map(|parent_uid| Self::parse_entity_uid(parent_uid, filename))
+            .collect::<Result<HashSet<_>, _>>()?;
+
+        // Use Cedar's from_json_value to parse the entity with attributes
+        // This properly handles attribute conversion to RestrictedExpression
+        let entity_json = serde_json::json!([{
+            "uid": {
+                "type": raw_entity.uid.entity_type,
+                "id": raw_entity.uid.id
+            },
+            "attrs": raw_entity.attrs,
+            "parents": raw_entity.parents
+        }]);
+
+        // Parse with optional schema validation
+        let entities_store = Entities::from_json_value(entity_json, schema).map_err(|e| {
+            PolicyStoreError::CedarEntityError {
+                file: filename.to_string(),
+                err: CedarEntityErrorType::JsonParseError(format!(
+                    "Failed to parse entity{}: {}",
+                    if schema.is_some() {
+                        " (schema validation failed)"
+                    } else {
+                        ""
+                    },
+                    e
+                )),
+            }
+        })?;
+
+        // Extract the single entity
+        let entity = entities_store
+            .iter()
+            .next()
+            .ok_or_else(|| PolicyStoreError::CedarEntityError {
+                file: filename.to_string(),
+                err: CedarEntityErrorType::NoEntityFound,
+            })?
+            .clone();
+
+        Ok(ParsedEntity {
+            entity,
+            uid,
+            filename: filename.to_string(),
+            content: content.to_string(),
+        })
+    }
+
+    /// Parse multiple entities from a JSON array or object.
+    ///
+    /// Supports both array format: `[{entity1}, {entity2}]`
+    /// And object format: `{"entity_id1": {entity1}, "entity_id2": {entity2}}`
+    pub fn parse_entities(
+        content: &str,
+        filename: &str,
+        schema: Option<&Schema>,
+    ) -> Result<Vec<ParsedEntity>, PolicyStoreError> {
+        let json_value: JsonValue =
+            serde_json::from_str(content).map_err(|e| PolicyStoreError::JsonParsing {
+                file: filename.to_string(),
+                source: e,
+            })?;
+
+        match json_value {
+            JsonValue::Array(arr) => {
+                // Array of entities
+                let mut parsed_entities = Vec::with_capacity(arr.len());
+                for (_idx, entity_json) in arr.iter().enumerate() {
+                    let entity_str = serde_json::to_string(entity_json).map_err(|e| {
+                        PolicyStoreError::JsonParsing {
+                            file: filename.to_string(),
+                            source: e,
+                        }
+                    })?;
+                    let parsed = Self::parse_entity(&entity_str, filename, schema)?;
```

Comment:
And we can avoid converting `entity_json` to string:

```suggestion
                    let entity = Self::parse_entity(entity_json, filename, schema)?;
```